fastestimator/trace/trace.py

```diff
@@ -13,36 +13,40 @@
 # limitations under the License.
 # ==============================================================================
 import time
-from typing import Any, Callable, Iterable, List, Union
+from typing import Any, Callable, Iterable, List, Union, Set
 
 import numpy as np
-
 from fastestimator.backend.to_number import to_number
-from fastestimator.util.util import to_list
+from fastestimator.util.system import System
+from fastestimator.util.util import to_list, draw
 
 
 class Trace:
     """Trace controls the training loop. User can use `Trace` to customize their own operations.
     Args:
         inputs: A set of keys that this trace intends to read from the state dictionary as inputs
-        outputs: A set of keys that this trace intends to write into the state dictionary
+        outputs: A set of keys that this trace intends to write into the system buffer
         mode: Restrict the trace to run only on given modes ('train', 'eval'). None will always
                         execute
     """
+    system: System
+
     def __init__(self,
                  inputs: Union[None, str, Iterable[str], Callable] = None,
-                 mode: Union[None, str, Iterable[str]] = None,
-                 log_names=None):
+                 outputs: Union[None, str, Iterable[str]] = None,
+                 mode: Union[None, str, Iterable[str]] = None):
         if isinstance(inputs, Iterable) and not isinstance(inputs, str):
```

Comment:
why not just call `to_list`

Reply:
cause we want single inputs to not be wrapped in list right?